pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.kt

```diff
@@ -127,25 +126,253 @@ class PushPayloadHelper {
                     if (it.isNotEmpty()) {
                         put("data", it)
                     }
                 }
                 notification?.let {
-                    it.toMap().run {
-                        if (this.isNotEmpty()) {
-                            put("notification", this)
+                    it.toMap().let { map ->
+                        if (map.isNotEmpty()) {
+                            put("notification", map)
                         }
                     }
                 }
+                android?.let {
+                    it.toMap().let { map ->
+                        if (map.isNotEmpty()) {
+                            put("android", map)
+                        }
+                    }
+                }
+                webpush?.let {
+                    it.toMap().let { map ->
+                        if (map.isNotEmpty()) {
+                            put("webpush", map)
+                        }
+                    }
+                }
+                apns?.let {
+                    it.toMap().let { map ->
+                        if (map.isNotEmpty()) {
+                            put("apns", map)
+                        }
+                    }
+                }
+                fcmOptions?.let {
+                    it.toMap().let { map ->
+                        if (map.isNotEmpty()) {
+                            put("fcm_options", map)
+                        }
+                    }
+                }
+            }
+        }
+
+        class WebpushConfig : PushPayloadSerializer {
+            var headers: Map<String, String>? = null
+            var data: Map<String, String>? = null
+            var notification: Map<String, Any>? = null
+            var fcmOptions: WebpushFcmOptions? = null
+
+            class WebpushFcmOptions : PushPayloadSerializer {
+                var link: String? = null
+                var analyticsLabel: String? = null
+
+                override fun toMap(): Map<String, Any> {
+                    return buildMap {
+                        link?.let { put("link", it) }
+                        analyticsLabel?.let { put("analytics_label", it) }
+                    }
+                }
+            }
+
+            override fun toMap(): Map<String, Any> {
+                return buildMap {
+                    headers?.let { put("headers", it) }
+                    data?.let { put("data", it) }
+                    notification?.let { put("notification", it) }
+                    fcmOptions?.let { put("fcm_options", it.toMap()) }
+                }
+            }
+        }
+
+        class ApnsConfig : PushPayloadSerializer {
+            var headers: Map<String, String>? = null
+            var payload: Map<String, String>? = null
+            var fcmOptions: ApnsFcmOptions? = null
+
+            class ApnsFcmOptions : PushPayloadSerializer {
+                var analyticsLabel: String? = null
+                var image: String? = null
```

Comment:
In this page there is none "image". How do you know that it is required?

Reply:
you mean in https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#apnsfcmoptions ? I see an optional image there

Reply:
Thanks for clarification.